google/oauth2/_client.py

```diff
@@ -101,7 +101,7 @@ def _token_endpoint_request(request, token_uri, body):
     }
 
     response = request(
-        method='POST', url=token_uri, headers=headers, body=body)
+        method='POST', url=token_uri, headers=headers, body=body, timeout=10)
```

Comment:
Why the timeout here? It seems arbitrary and by default it'll match the default socket timeout iirc.

Reply:
> it'll match the default socket timeout

Unfortunately no. It depends on HTTP client libraries.

> It seems arbitrary

I think so too, and want some suggestion.

`requests` doesn't respect `socket.getdefaulttimeout()` nor have its own timeout value, at least of versions lower than 3.0.0.

[requests doc says](http://docs.python-requests.org/en/master/user/quickstart/#timeouts)

> If no timeout is specified explicitly, requests do not time out.

There are many issues discuss this problematic behaviour, such as

 * https://github.com/requests/requests/issues/3222
 * https://github.com/shazow/urllib3/issues/655

So, what timeout value would you prefer to?
I think it is nice that this library has its own default timeout value, not relying on HTTP libraries' defaults.

If you choose `socket.getdefaulttimeout()`, The users of `google-auth` must call `socket.setdefaulttimeout()` in the their code, which seems to me somewhat ugly.

Also, I think you can set more shorter timeout value with knowledge about mechanisms of authentication methods this library supports.
It seems to me too conservative to use HTTP libraries' timeout value.
Are there any cases where it take 30 sec. or so to refresh an OAuth token on server side afterall?
Like we don't set the same timeout value to HTTP requests with heavy tasks such as SQL slow queries and ones that render simple HTML pages, we could set a timeout value more suited for the work this library does.